[CI] RRFRetrieverBuilderNestedDocsIT testRRFRetrieverPartialSearchErrorsTrue failing

**Build Scans:**
- [elasticsearch-intake #40886 / part2](https://gradle-enterprise.elastic.co/s/hvv74nqgrrgsa)
- [elasticsearch-pull-request #152305 / part-2](https://gradle-enterprise.elastic.co/s/f2g4uust34zxo)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:rank-rrf:internalClusterTest" --tests "org.elasticsearch.xpack.rank.rrf.RRFRetrieverBuilderNestedDocsIT.testRRFRetrieverPartialSearchErrorsTrue" -Dtests.seed=ACFAFEA2BBD6AE37 -Dtests.locale=en-GH -Dtests.timezone=Australia/Broken_Hill -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.rank.rrf.RRFRetrieverBuilderNestedDocsIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testRRFRetrieverPartialSearchErrorsTrue'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: safeGet: listener was not completed within the timeout
```

**Issue Reasons:**
- [main] 2 consecutive failures in test testRRFRetrieverPartialSearchErrorsTrue
- [main] 1 consecutive failures in step part2
- [main] 1 consecutive failures in step part-2
- [main] 1 consecutive failures in pipeline elasticsearch-intake
- [main] 1 consecutive failures in pipeline elasticsearch-pull-request
- [main] 2 failures in test testRRFRetrieverPartialSearchErrorsTrue (0.5% fail rate in 402 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

The test fail in the clean up step
```
listener was not completed within the timeout
	at org.elasticsearch.test.ESTestCase.safeGet(ESTestCase.java:2816)
	at org.elasticsearch.test.ESTestCase.safeGet(ESTestCase.java:2797)
	at org.elasticsearch.test.ESIntegTestCase.doEnsureClusterStateConsistency(ESIntegTestCase.java:1806)
	at org.elasticsearch.test.ESIntegTestCase.ensureClusterStateConsistency(ESIntegTestCase.java:1708)
	at org.elasticsearch.test.ESIntegTestCase.afterInternal(ESIntegTestCase.java:639)
	at org.elasticsearch.test.ESIntegTestCase.cleanUpCluster(ESIntegTestCase.java:2967)
```

The timeout was caused by an assertion failure

```	
        at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotNull(Assert.java:756)
	at org.junit.Assert.assertNull(Assert.java:738)
	at org.elasticsearch.test.ESIntegTestCase.lambda$doEnsureClusterStateConsistency$26(ESIntegTestCase.java:1788)
```

With the message 
```
java.lang.AssertionError: cluster state JSON serialization does not match expected null, but was:&lt;/routing_nodes/unassigned/*/unassigned_info/details: the elements don't match: [failed shard on node [Sn9vaEymSOSIH20M1BmbRQ]: shard failure, reason [search execution corruption failure], failure [test_index/L0DYwI4BQL-ScjGBRuxIMA] org.elasticsearch.index.query.QueryShardException: failed to create query: simulated failure (resource=null)
```

Seems to be due to a random shard failure

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -524,9 +524,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.analysis.AnalyzerUnmappedGoldenTests
   method: testTypeConflictTimeseriesDoubleUnmappedWithCast
   issue: https://github.com/elastic/elasticsearch/issues/150672
-- class: org.elasticsearch.xpack.rank.rrf.RRFRetrieverBuilderNestedDocsIT
-  method: testRRFRetrieverPartialSearchErrorsTrue
-  issue: https://github.com/elastic/elasticsearch/issues/150673
 - class: org.elasticsearch.xpack.remotecluster.CrossClusterEsqlRCS1EnrichUnavailableRemotesIT
   method: testEsqlEnrichWithSkipUnavailable
   issue: https://github.com/elastic/elasticsearch/issues/150686

--- a/x-pack/plugin/rank-rrf/src/internalClusterTest/java/org/elasticsearch/xpack/rank/ShardFailingQueryBuilder.java
+++ b/x-pack/plugin/rank-rrf/src/internalClusterTest/java/org/elasticsearch/xpack/rank/ShardFailingQueryBuilder.java
@@ -7,9 +7,7 @@
 
 package org.elasticsearch.xpack.rank;
 
-import org.apache.lucene.index.CorruptIndexException;
 import org.apache.lucene.search.Query;
-import org.apache.lucene.store.DataInput;
 import org.elasticsearch.TransportVersion;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;

--- a/x-pack/plugin/rank-rrf/src/internalClusterTest/java/org/elasticsearch/xpack/rank/ShardFailingQueryBuilder.java
+++ b/x-pack/plugin/rank-rrf/src/internalClusterTest/java/org/elasticsearch/xpack/rank/ShardFailingQueryBuilder.java
@@ -58,7 +58,7 @@ public class ShardFailingQueryBuilder extends LeafQueryBuilder<ShardFailingQuery
     @Override
     protected Query doToQuery(SearchExecutionContext context) throws IOException {
         if (context.getShardId() % 2 == 0) {
-            throw new CorruptIndexException("simulated failure", (DataInput) null);
+            throw new IllegalArgumentException("simulated failure");
         } else {
             return Queries.ALL_DOCS_INSTANCE;
         }

